giftless/storage/azure.py

```diff
@@ -124,11 +142,19 @@ def get_multipart_actions(self, prefix: str, oid: str, size: int, part_size: int
             }
         }
 
+        if filename:
+            mime_type = self._guess_mime_type_from_filename(filename)
+            if mime_type:
+                reply["actions"]["commit"]["header"]["x-ms-blob-content-type"] = mime_type
+
         if parts:
             reply['actions']['parts'] = parts
 
         return reply
 
+    def _guess_mime_type_from_filename(self, filename: str) -> Optional[str]:
```

Comment:
It might be useful to move this to somewhere more common e.g. `util` or the `giftless.storage` module `__init__`, because I have a feeling this won't be Azure specific. If not, we can make it a `@staticmethod`.

Reply:
I moved it to a new `BaseStorage` class to not mix it with `VerifiableStorage` but let me know if you prefer an alternative.

Unrelated question, is it necessary for the different `*Storage` classes to extend from `VerifiableStorage` and `ABC` when `VerifiableStorage`  extends `ABC`?

Reply:
Since I don't see this as part of the class hierarchy, I would prefer to put this in a separate utility function. 

The only reason to extend `ABC` is to designate a class as abstract and not a concrete implementation of a storage backend / adapter. So I used `ABC` to mark any abstract storage interface, that users shouldn't instantiate directly - even if it already inherits from `VerifiableStorage`, if it is abstract I marked it as a subclass of `ABC`. I don't think it's *necessary*, I just thought it would make it easier to distinguish between storage backends that are concrete and ones that just designate a certain interface.